app/assets/javascripts/discourse/app/mixins/uppy-upload.js

```diff
@@ -119,6 +120,13 @@ export default Mixin.create(UppyS3Multipart, {
           this._reset();
           return false;
         }
+
+        // for a single file, we want to override file meta with the
+        // data property (which may be computed), to override any keys
+        // specified by this.data (such as name)
+        if (this.data && !isEmpty(this.data) && fileCount === 1) {
```

Comment:
Docs on `isEmpty`:

> Verifies that a value is null or undefined, an empty string, or an empty array.
> isEmpty({});               // false

Reply:
Wow that's what I get for making assumptions! Will fix